### DISCUSSION TITLE
Messaging/Notifications Integration

### DISCUSSION THREAD

**Discussion by cryptoKevinL** [MetaMask/snaps#670] [Ideas]
Hi Snaps team and devs,

I have a project called WalletChat which does wallet-to-wallet messaging and notifications via Chrome Extension.  I see this is one area of focus requested by the Metamask snaps teams.  However, without UI changes the messaging app is nothing more than me displaying fetched data from our API in the snaps window.  

If we could use alarms (from Chrome Extension framework) in MM Snaps however, we could alert users via Badge notifications of new messages - which seems quite useful.  Obviously spam filtering would be something which would need focus for any open messaging solution, but thats its own effort. Will alarms be supported in MM snaps eventually (or maybe it's possible somehow today)?

We look forward to helping integrate this into MM snaps and see what messaging can become integrated with a wallet as trusted as Metamask.

**Comment by rekmarks** [comment]
Hi @cryptoKevinL, that sounds like an interesting use case. First, a clarifying question: Is your intention to use `chrome.alarms` to essentially schedule a job that checks for new messages an updates the notification badge?

**Reply by cryptoKevinL** [reply]
Hi @rekmarks, you are right on.  Currently our Chrome Extension uses Manifest V3, and uses chrome.alarms to wake up and fetch data from an API, and set notifications.  This is particularly useful since it runs with the extension closed/not in view.  This way users can see a message is waiting in a very non-intrusive way.  I assume at this early stage Snaps doesn't support this yet, but I'm happy to share our code for this with you directly if needed.  Even though Snaps doesn't allow custom UI changes yet, this would still be really useful because people could use the WalletChat web app to send and read messages, but still only need Metamask installed to get their trusted source of notifications.
